libs/community/langchain_community/chat_models/oci_generative_ai.py

```diff
@@ -158,7 +158,10 @@ def chat_response_to_text(self, response: Any) -> str:
 
     def chat_stream_to_text(self, event_data: Dict) -> str:
         if "text" in event_data:
-            return event_data["text"]
+            if "finishedReason" in event_data or "toolCalls" in event_data:
+                return ""
+            else:
+                return event_data["text"]
```

Comment:
```suggestion
            return event_data.get("text")
```

Reply:
@efriis It is actually not a key error. It is the mechanism of our API. When finishedReason and toolCalls exists in the response, "text" field will output all the previous content generated and combine them together. This will make the streaming result to be duplicated two to three times.